### PR TITLE
One hot encoding added and encoding module refractored

### PR DESCRIPTION
**Pull request by Bhumika-Kothwal**
Solves issue #32 
Solves issue #43 

The files refractored are : 

- preprocessy/encoding/_categorical.py
- preprocessy/encoding/test.py
- tests/test_encode.py

New function added : 
- __encode_one_hot_util() function

### CODE REVIEW

File: preprocessy/encoding/_categorical.py [hunk 1]
```diff
@@ -91,47 +68,78 @@ def __encode_categorical_util(self):
...
+                    elif (
+                        isinstance(self.train_df[col][0], int)
+                        and self.train_df[col].nunique() < rows
+                    ):
+                        self.cat_cols.append(col)
+                    elif (
+                        isinstance(self.train_df[col][0], str)
+                        and self.train_df[col].nunique() < rows
+                    ):
+                        self.cat_cols.append(col)
```
Comment (Saif807380) [inline preprocessy/encoding/_categorical.py:135]: You can combine these in one - 

```python

elif type(self.train_df[col][0]) in ["int", "str"]:
```

File: preprocessy/encoding/_categorical.py [hunk 2]
```diff
@@ -91,47 +68,78 @@ def __encode_categorical_util(self):
...
+                else:
+                    continue
 
+        if self.one_hot:
+            self.__encode_one_hot_util()
+        else:
+            self.__encode_categorical_util()
         return True
```
Comment (Saif807380) [inline preprocessy/encoding/_categorical.py:143]: No need to return True here. An empty return statement will be fine.

File: tests/test_encode.py [hunk 3]
```diff
@@ -5,40 +5,57 @@
...
+            "target_label": "Price",
+            "cat_cols": ["Professions", "Test", "Labels"],
+            "ord_dict": ord_dict,
+            "one_hot": True,
+        }
+        encoder = EncodeData()
+        encoder.encode(params=params)
+        assert "Test_Tata" in params["train_df"].columns
+        assert params["train_df"]["Test_Tata"][1] == 1
+        assert "Profession_HOD" not in params["train_df"].columns
```
Comment (Saif807380) [inline tests/test_encode.py:61]: Can you put this in a separate test? It was kind of hard to spot what assertion implies.

**Review by Saif807380 — CHANGES_REQUESTED** [review]
Great work. Just a few changes - 

- You can delete `preprocessy/encoding/test.py`

File: preprocessy/encoding/_categorical.py [hunk 4]
```diff
@@ -152,14 +160,41 @@ def __encode_ordinal(self):
...
+            train_df    : training dataset (pandas.core.frames.DataFrame)
+            test_df     : test dataset (pandas.core.frames.DataFrame)
+            target_label: name of target column so as to ensure that target columns isn't present in
+                          train_df
+        }
+
+        """
 
-    def encode(self):
         """
```
[the remark this answers is not in the document]
Comment (Bhumika-Kothwal) [reply preprocessy/encoding/_categorical.py:191]: Okay will do the change.

File: preprocessy/encoding/_categorical.py (hunk 1 above)
Comment (Bhumika-Kothwal) [reply preprocessy/encoding/_categorical.py:135]: Sure will do this change.

File: preprocessy/encoding/_categorical.py (hunk 2 above)
Comment (Bhumika-Kothwal) [reply preprocessy/encoding/_categorical.py:143]: Okay will change this.

**Review by Saif807380 — APPROVED** [review]
Merging this. Good Work.

**Pull request merged**